### ISSUE TITLE
Correct styling is failing when using nested media queries in SCSS

### ISSUE THREAD

**Issue by engelfrost** [YozhikM/stylelint-a11y#38]
I'm using the media-prefers-reduced-motion rule, and am having the following issue with the linter: 

```scss
/* Fails */
.foo {
  transition: all;

  @media (prefers-reduced-motion: reduce) {
    transition: none;
  }
}
```

```css 
/* Works */
.foo {
  transition: all; 
}

@media (prefers-reduced-motion: reduce) {
  .foo {
    transition: none;
  }
}
```

The SCSS in the first code snippet produces the CSS in the second code snippet, yet the first example fails while the second example succeeds.

**Comment by stacylondon** [comment]
I'm running into a similar issue using Emotion (css-in-js) and specifically `@emotion/styled`.  Here is an example of where the media queries are generated correctly and apply properly when I test via System Preferences > Accessibility > Display > and check “Reduced motion” but the linter is still flagging an issue:

```javascript
export const MyStyle = styled.div`
  transition: all;

  @media screen and (prefers-reduced-motion: reduce) {
    transition: none;
  }
`;
```

**Comment by evilebottnawi** [comment]
Yes, it is bug, PR welcome
